components/blitz/resources/omero/api/IContainer.ice

```diff
@@ -19,25 +19,420 @@ module omero {
+                 *            filtering will be applied at the
+                 *            <i>Class</i>-level; e.g. to retrieve a user's
+                 *            Projects, or user's Datasets. If
+                 *            <code>rootNodeIds!=null</code>, the result will
+                 *            be filtered by the
+                 *            <code>experimenter|group</code> at the
+                 *            <code>Image</code> and intermediate levels
+                 *            <i>if available</i>.
+                 *            Due to the amount of data potentially linked a
+                 *            Screen/Plate, the <code>leaves</code> option is
+                 *            not taken into account when the root node is a
+                 *            {@link omero.model.Screen}.
+                 *            <code>orphan</code> implies that objects which
+                 *            are not contained in an object of rootNodeType
+                 *            should also be returned.
+                 * @return a set of hierarchy trees. The requested node as
+                 *         root and all of its descendants. The type of the
+                 *         returned value will be <code>rootNodeType</code>,
+                 *         unless <code>orphan</code> is specified in which
+                 *         case objects of type <code>rootNodeType</code> and
+                 *         below may be returned.
+                 */
                 idempotent IObjectList loadContainerHierarchy(string rootType, omero::sys::LongList rootIds, omero::sys::Parameters options) throws ServerError;
+
+                /**
+                 * Retrieves hierarchy trees in various hierarchies that
+                 * contain the specified Images.
+                 * <p>
+                 * This method will look for all the containers containing the
+                 * specified Images and then for all containers containing
+                 * those containers and on up the container hierarchy.
+                 * </p>
+                 * <p>
+                 * This method returns a <code>Set</code> with all root nodes
+                 * that were
+                 * found. Every root node is linked to the found objects and
+                 * so on until the leaf nodes, which are
+                 * {@link omero.model.Image} objects. Note that the type of any
+                 * root node in the returned set can be the given
+                 * rootNodeType, any of its containees or an
+                 * {@link omero.model.Image}.
+                 * </p>
+                 * <p>
+                 * For example, say that you pass in the ids of six Images:
+                 * <code>i1, i2, i3, i4, i5, i6</code>.
+                 * If the P/D/I hierarchy in the DB looks like this:
+                 * </p>
+                 *
+                 * <pre>
```

Comment:
The <pre> block in the rendered doc is not well formatted, see: https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/slice2html/omero/api/IContainer.html#findContainerHierarchies